lti_consumer/plugin/views.py

```diff
@@ -111,8 +117,9 @@ def public_keyset_endpoint(request, usage_id=None):
         response = JsonResponse(lti_config.lti_1p3_public_jwk)
         response['Content-Disposition'] = 'attachment; filename=keyset.json'
         return response
-    except (LtiError, InvalidKeyError, ObjectDoesNotExist):
-        return HttpResponse(status=404)
+    except (LtiError, InvalidKeyError, ObjectDoesNotExist) as exc:
+        log.info("Error while retrieving keyset for given usage_id.")
```

Comment:
Why is this `info` if it is an error?  

More importantly, and throughout all the logging: we should include data in the message.  What usage_id? The message from the exception would be helpful:
```
log.info("Error while retrieving keyset for usage_id %r: %s", usage_id, exc)
```

Reply:
@nedbat 
>Why is this info if it is an error?

These are known errors that can happen during normal usage of the LTI block, so that's why I didn't want to fill the logs with errors related to normal usage most of the time. I'll change them to `exception` again since our goal is to make the tool easy to debug.

> More importantly, and throughout all the logging: we should include data in the message. 

Makes sense, I'll improve the logs and add more data.

Reply:
You know better than me how likely the exception is.  If this is something that a site operator should be concerned about, then we should make it a warning.  If it's something that can happen because an instructional team made a typo, then we can leave it as `info`.

Reply:
I've replaced some instances of `info` with `warning` in permission denied situations and others where the error handling is done inside the XBlock.

I've also improved the error messages, and checked that the exceptions inside the LTI 1.3 module pass down the full traceback (`except ... as exc: raise ... from exc`).